2048/2048/src/main/java/com/appcoins/eskills2048/rankins/RankingsPagerAdapter.java

```diff
@@ -0,0 +1,52 @@
+package com.appcoins.eskills2048.rankins;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.viewpager2.adapter.FragmentStateAdapter;
+import com.appcoins.eskills2048.R;
+import com.appcoins.eskills2048.repository.StatisticsTimeFrame;
+import java.security.InvalidParameterException;
+
+class RankingsPagerAdapter extends FragmentStateAdapter {
+
+  private final String walletAddress;
+
+  public RankingsPagerAdapter(@NonNull Fragment fragment, String walletAddress) {
+    super(fragment);
+    this.walletAddress = walletAddress;
+  }
+
+  @NonNull @Override public Fragment createFragment(int position) {
+    return RankingsContentFragment.newInstance(walletAddress, getTimeFrame(position));
+  }
+
+  private StatisticsTimeFrame getTimeFrame(int position) {
+    switch (position) {
+      case 0:
+        return StatisticsTimeFrame.TODAY;
+      case 1:
+        return StatisticsTimeFrame.WEEK;
+      case 2:
+        return StatisticsTimeFrame.ALL_TIME;
+      default:
+        throw new InvalidParameterException("Invalid position " + position);
+    }
+  }
+
+  @Override public int getItemCount() {
+    return 3;
```

Comment:
this 3 is related to how many items are inside StatisticsTimeFrame, right? why just not: StatisticsTimeFrame.values().length